frontend/src/metabase/account/profile/components/UserProfileForm/UserProfileForm.jsx

```diff
@@ -16,7 +16,9 @@ const UserProfileForm = ({ user }) => {
     [user],
   );
 
-  return <User.Form user={user} form={User.forms.user} onSaved={handleSaved} />;
+  return (
+    <User.Form user={user} form={User.forms.user(user)} onSaved={handleSaved} />
+  );
```

Comment:
One thing you can potentially do here is to use the form framework's `form.fields` function notation:

```jsx
// Somewhere around metabase/entities/users
const userForm = {
   // Form framework will pass the user object by it's own
   fields: user => {
      const isSsoUser = check(user)
      ...
      return { fields: [ ... ], ...formProps }
   }
}

// UserProfileForm, notice we're just passing the form object, without passing the user object
return <User.Form user={user} form={User.forms.user} onSaved={handleSaved} />;
```

`form.fields` can be a function and our `Form` framework will pass the user action as the first argument, so we can potentially avoid passing it right here and in any other user form later. But it's worth checking it receives the complete `user` object so you have everything needed to figure out if a user had signed up with SSO

Reply:
@kulyk This doesn't work. There's nothing passes to `fields` function. I suspect because `user` I'm using here is from  the current user which is another endpoint (not the users endpoint)

https://github.com/metabase/metabase/blob/ae26bb6e60a6baab29bafc0903f8c2414cfa167a/frontend/src/metabase/account/profile/containers/UserProfileApp/UserProfileApp.jsx#L5-L7

https://github.com/metabase/metabase/blob/d0f3fea4d8cafa8b2ef666d2ca44334ebf9bb239/frontend/src/metabase/selectors/user.js#L4

In short, it's unfortunate that I can't remove the use of `user` here. I'll keep the change as is.

Reply:
@ranquild Yeah that seems like so, so maybe I did something wrong I'll try that again.

Reply:
@WiNloSt I'm fine with how it is fwiw, just wanted to mention that `fields` function should also work

Reply:
@ranquild Got it. It's also cleaner so I wish it works.

Reply:
👍